chanutils/queue.go

```diff
@@ -0,0 +1,132 @@
+package chanutils
+
+import (
+	"container/list"
+	"sync"
+)
+
+const (
+	// DefaultQueueSize is the default size to use for concurrent queues.
+	DefaultQueueSize = 10
+)
+
+// ConcurrentQueue is a typed concurrent-safe FIFO queue with unbounded
+// capacity. Clients interact with the queue by pushing items into the in
+// channel and popping items from the out channel. There is a goroutine that
+// manages moving items from the in channel to the out channel in the correct
+// order that must be started by calling Start().
+type ConcurrentQueue[T any] struct {
+	started sync.Once
+	stopped sync.Once
+
+	chanIn   chan T
+	chanOut  chan T
+	overflow *list.List
```

Comment:
Can use this generic list to inherit type safety (or sort of push down the concerns to another layer): https://github.com/lightninglabs/neutrino/blob/5aac983d75cc02fce159b592dab7940df7d494a7/cache/lru/list.go#L62-L63

Reply:
oh sweet - that is perfect. Updated!